babyai/bot.py

```diff
@@ -317,154 +317,6 @@ def take_action(self, action):
-                    elif self.fwd_cell.type.endswith('door') and self.fwd_cell.is_open:
-                        # add a useless subgoal that will force unblocking
-                        self.bot.stack.append(GoNextToSubgoal(self.bot, self.fwd_pos + 2 * self.dir_vec))
-                        return True
-            else:
-                # If we are right in front of the object, then the subgoal is completed
-                # go back to the previous subgoal
-                if np.array_equal(obj_pos, self.fwd_pos):
-                    self.bot.stack.pop()
-                    return False
-
-
-            # CASE 1.2: The action will make us in front of the object
-            if np.array_equal(obj_pos, self.new_fwd_pos):
-                assert action in (self.actions.forward, self.actions.left, self.actions.right), "Something seems wrong"
-                # stack will be popped at next step
-                return True
-                # It is impossible that it's another action that yields this CASE 1.2
-
-            # CASE 1.3: There is a blocking object in front of us, and we aren't carrying anything
-            if self.fwd_cell is not None and not self.fwd_cell.type.endswith('door') and self.carrying is None:
-                if action == self.actions.pickup:
-                    # -> Create a GoNextTo subgoal
-                    # TODO: I am not sure this is enough. Shouldn't we drop the object somewhere?
-                    self.bot.stack.append(GoNextToSubgoal(self.bot, obj_pos))
-                    # TODO: Double check that False should be returned here
-                    return False
-
-            # CASE 1.4: All other scenarios
-            if action in (self.actions.drop, self.actions.pickup, self.actions.toggle):
-                # Update the stack if we did something bad, or do nothing if the action doesn't change anything
-                self.foolish_action_while_exploring(action)
-                return True
-
-            if action == self.actions.done or (self.fwd_cell is not None and action == self.actions.forward):
-                # Do nothing if the action doesn't change anything
-                return True
-
-            # Otherwise, the taken action (forward, left or right) has necessarily changed the agent state
-            # -> Create a GoNextTo subgoal
-            else:
-                path, _, _ = self.bot.shortest_path(
-                    lambda pos, cell: pos == obj_pos,
-                    try_with_blockers=True
-                )
-
-                if path:
-                    # New subgoal: go next to the object
-                    self.bot.stack.append(GoNextToSubgoal(self.bot, obj_pos, reason='GoToObj', adjacent=self.adjacent))
-                    return True
-
-        # CASE 2: The bot hasn't seen the object. The action will count for the new subgoal
-        self.bot.stack.append(ExploreSubgoal(self.bot))
-        return False
-
-
 class GoNextToSubgoal(Subgoal):
```

Comment:
Can you add a class-level comment explaining what this subgoal does in some detail?

Reply:
I wrote smth, let me know if it is understandable. I don't yet understand all of the flags though, for them I just copied the docstring that Salem had written.

Reply:
Looks good